src/main/cli/process.ts

```diff
@@ -0,0 +1,133 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { app, dialog } from "electron";
+import * as glob from "glob";
+import * as path from "path";
+import { _PACKAGING } from "readium-desktop/preprocessor-directives";
+import * as yargs from "yargs";
+
+import { cli_, cliImport, cliOpds, cliRead } from "./commandLine";
+
+declare const __APP_NAME__: string;
+declare const __APP_VERSION__: string;
+
+// Logger
+const debug = debug_("readium-desktop:cli");
+
+export function cli(mainFct: () => void) {
+
+    yargs
+        .scriptName(__APP_NAME__)
+        .version(__APP_VERSION__)
+        .usage("$0 <cmd> [args]")
+        .command("opds <title> <url>",
+            "import opds feed",
+            (y) =>
+                y.positional("title", {
+                    describe: "title opds feed",
+                    type: "string",
+                })
+                .positional("url", {
+                    describe: "url opds feed",
+                    type: "string",
+                })
+            ,
+            (argv) => {
+                const promise = cliOpds(argv.title, argv.url);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return;
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("import <path>",
+            "import epub or lpcl file",
+            (y) =>
+                y.positional("path", {
+                    describe: "absolute, relative or globbing path",
+                    type: "string",
+                })
+                .example("import", "\"./myPath/**/*.epub\"")
+                .example("import", "\"/*.epub\"")
+                .example("import", "\"myPublication.epub\"")
+            ,
+            (argv) => {
+                const pathArray = glob.sync(argv.path, {
+                    absolute: true,
+                    realpath: true,
+                }) || [];
+                const promise = cliImport(pathArray);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return;
+                    }
+                    app.exit(1);
+                }).catch(() => {
```

Comment:
Suggestion: `process.stderr.write()`of the caught promise rejection (need to pass the `err` param in the `.catch()` callback).

This caught error can (for example) be the *uncaught* promise rejection from `if (!await catalogService.importFile(fp))` which is called inside the `cliImport()` function. This is why I suggest reporting the full error context to the CLI user.

Reply:
done